### ISSUE TITLE
documentation

### ISSUE THREAD

**Issue by spyroot** [reductstore/reductstore#369]
Hi Folks,

1)
I opened a separate issue since it is neither a bug nor a feature.   Please indicate the recommended hierarchy and how you would want to store the data.  I tried to keep data as entry name 2023_10_22/20_33_39.jpg and noticed that reduct.error.ReductError: Status 405: Unknown error constantly returns errors; hence, I assume the recommended approach stores data in a flat structure,  and path /something/data is not supported. 

Then I relate that Flat Storage Structure. It doesn't have a tree-like structure for data. There are only buckets and entries with unique names in them.

My initial assumption was that you don't have a tree-like structure internally, but from a naming convention, I can use a name that contains a tree-like structure. (i.e., in the end, it is just a name)

2) Could you also indicate what is a trade-off by introducing many buckets? In essence, if the system provides only flat storage and you do need some hierarchy, can you use buckets for that?

**Comment by atimin** [comment]
Hi @spyroot, thank you for the report. I'll update the documentation and add details about the data structure soon. However, you can imitate a tree structure in the following ways:

1. Prefixes in entry names. I think this is the easiest way to structure the data:
```
bucket
  |------ folder_A__entry_A
  |-------folder_A__entry_B
  |-------folder_B__entry_C
```
I don't see any disadvantages to this method, but your code should handle the names.

2. Buckets. ReductStore doesn't have any restrictions on the number of buckets. So you can use them as an additional level of hierarchy.
```
bucket_A
|------entry_A
|------entry_B

bucket_B
|-----entry_C
```
The disadvantage is that a bucket is not only a container for entries, it has settings and is used for access control. I would suggest using separate buckets if you need to separate access to your data or have different quotas. The SDK and CLI tools are also bucket-centric. However, you will be safer in the future by using many buckets because you can organize a load balancing between them by using a reverse-proxy.

3. Labels. You can label data in ReductStore with a folder name:
```
bucket
|-----entry_A [folder=A]
|-----entry_B [folder=A]
|-----entry_C [folder=B]
```
When your data is labeled, you can use the `include` option to filter records by folder. This may be a very convenient way, but a label is attached to each record, so you will have some little overhead. 

I hope this will help you. 

PS. your example of an entry name  (`2023_10_22/20_33_39.jpg `) makes me suspicious that you are trying to organize data partitioning by time. You may need it in some S3-like blob storage, but not in ReductStore. It does the partitioning out of the box in a very efficient way.

**Comment by atimin** [comment]
Addressed here reductstore/website#33
